src/rust/ide/view/src/searcher/icons.rs

```diff
@@ -0,0 +1,387 @@
+// To compensate for this, we apply `.shrink(SHRINK_AMOUNT)` to all icons in this file. By default,
+// `SHRINK_AMOUNT` should be set to 0.0 to make the icons appear right on the default zoom level.
+// When examining the icons closely, with a strong zoom, `SHRINK_AMOUNT` should be set to 0.4 to
+// make the icons apear right on that zoom level.
+const SHRINK_AMOUNT : f32 = 0.0;
+// const SHRINK_AMOUNT : f32 = 0.38;
+
+
+
+// =================
+// === Favorites ===
+// =================
+
+/// A five-pointed star.
+mod star {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+            // The distance from the center to the outer points.
+            let radius = 7.0;
+            // The ration between the radius and the distance from center to the points where spikes
+            // touch.
+            let ratio = 0.447;
+            // The angle between neighboring spikes.
+            let rotation_step = 2.0 * PI / 5.0;
+            // By "pentagon", we refer to the pentagon that forms the body of the star, filling the
+            // whole space in the middle, connecting the spikes.
+            let pentagon_radius       = ratio * radius;
+            let pentagon_inner_radius = pentagon_radius * (rotation_step / 2.0).cos();
+            let pentagon_side_length  = pentagon_radius * (rotation_step / 2.0).sin() * 2.0;
+            let spike_length          = radius - pentagon_inner_radius;
+
+            // The spike pointing upwards.
+            let spike = Triangle(pentagon_side_length.px(),spike_length.px());
+            let spike = spike.translate_y((radius - spike_length / 2.0).px());
+            // We compose the pentagon of five triangles fanning out from the center. This is one of
+            // them.
+            let pentagon_fragment = Triangle(pentagon_side_length.px(),pentagon_inner_radius.px());
+            let pentagon_fragment = pentagon_fragment.rotate(PI.radians());
+            let pentagon_fragment = pentagon_fragment.translate_y((pentagon_inner_radius/2.0).px());
+
+            // The spike and pentagon fragment form a star fragment. We copy it five times with
+            // rotations to form the whole star.
+            let fragment1 = spike + pentagon_fragment;
+            let fragment2 = fragment1.rotate((rotation_step * 1.0).radians());
+            let fragment3 = fragment1.rotate((rotation_step * 2.0).radians());
+            let fragment4 = fragment1.rotate((rotation_step * 3.0).radians());
+            let fragment5 = fragment1.rotate((rotation_step * 4.0).radians());
+
+            let shape = fragment1 + fragment2 + fragment3 + fragment4 + fragment5;
+            let shape = shape.fill(style.get_color(theme::favorites));
+            let shape = shape.translate_y((-0.5).px());
+            shape.shrink(SHRINK_AMOUNT.px()).into()
+        }
+    }
+}
+
+
+
+// ==========
+// === IO ===
+// ==========
+
+// A rounded rectangle with an arrow pointing in from the left.
```

Comment:
Could you use in docs a small ascii art? It will take 5 mins per comment and would increase the readability sooooo much, like:

```
╭──╮
│ <─
╰──╯
```

Reply:
I am a bit reluctant to do this. I would rather refer to the debug scene in the module documentation and maybe add labels to the scene to make it easy to find the corresponding icon for each shape definition. Would you be happy with that?

Reply:
Ok, lets skip it altogether. It would make the docs much nicer but yeah, lets just skip it.